src/components/SignInForm.js

```diff
@@ -0,0 +1,80 @@
+import { useState, useRef, useEffect, useContext } from 'react';
+import '../styles/signInForm.css';
+import axios from 'axios';
+import { UserContext } from '../UserContext';
+import { BASE_URL } from '../App';
+
+const LOGIN_URL = `${BASE_URL}/login`;
+const SignInForm = ({ closeSecondModal }) => {
+  const errRef = useRef();
+  const { setUser } = useContext(UserContext);
+  const [loginDetails, setLoginDetails] = useState({
+    username: '',
+    password: '',
+  });
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    setErrorMessage('');
+  }, [loginDetails.username, loginDetails.password]);
+
+  const handleChange = (e) => {
+    setLoginDetails({
+      ...loginDetails,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    try {
+      const { data } = await axios.post(LOGIN_URL, loginDetails);
+      setUser(data.user);
+      localStorage.setItem('loginDetails', JSON.stringify(data));
```

Comment:
If you are only using the user value, why are you storing the whole `data` object in `localStorage`?

Reply:
i also need to store the access token and refresh token

Reply:
Sir i have been running into an issue with my api calls i keep getting cors error on other browser except on my main chrome browser and even on my chrome browser i can only call the register and login endpoints i get cors error on other api calls

Reply:
The CORS error is for your backend to solve